TM1py/Utils/Utils.py

```diff
@@ -111,6 +111,122 @@ def build_content_from_cellset(raw_cellset_as_dict, cell_properties, top=None):
     return content_as_dict
 
 
+def build_arrays_from_cellset(raw_cellset_as_dict):
+    """ Transform raw 1,2 or 3-dimension cellset data into concise dictionary
+
+    :param raw_cellset_as_dict: raw data from TM1
+    :return: dict : { titles: [], headers: [axis][], cells: { Page0: { Row0: { [row values], Row1: [], ...}, ...}, ...} }
+    """
+    dimensionality = len(raw_cellset_as_dict['Axes'])
+    cardinality = [raw_cellset_as_dict['Axes'][axis]['Cardinality'] for axis in range(dimensionality)]
+
+    headerMap = build_headers_from_cellset(raw_cellset_as_dict)
+    titles = headerMap['titles']
+    headers = headerMap['headers']
+
+    # Handle 1, 2 and 3-dimensional cellsets. Use dummy row/page headers when missing
+    if len(headers) == 1:
+        headers += [[{'name':'Row'}]]
+        cardinality.insert(1,1)
+    if len(headers) == 2:
+        headers += [[{'name':'Page'}]]
+        cardinality.insert(2,1)
+
+    cells = {}
+    ordinal_cells = 0
+    for z in range(cardinality[2]):
+        zHeader = headers[2][z]['name']
+        pages = {}
+        for y in range(cardinality[1]):
+            yHeader = headers[1][y]['name']
+            row = []
+            for x in range(cardinality[0]):
+                row.append(raw_cellset_as_dict['Cells'][ordinal_cells]['Value'] or 0)
+                ordinal_cells += 1
+            pages[yHeader] = row
+        cells[zHeader] = pages
+
+    return {'titles':titles, 'headers':headers, 'cells':cells}
+
+def build_dygraph_arrays_from_cellset(raw_cellset_as_dict):
+    """ Transform raw 1,2 or 3-dimension cellset data into dygraph-friendly format
+
+    :param raw_cellset_as_dict: raw data from TM1
+    :return: dict : { titles: [], headers: [axis][], cells: { Page0: [  [column name, column values], [], ... ], ...} }
+    """
+    dimensionality = len(raw_cellset_as_dict['Axes'])
```

Comment:
Maybe outsource these two lines into seperate function to avoid code duplication:
dimensionality = len(raw_cellset_as_dict['Axes'])
cardinality = [raw_cellset_as_dict['Axes'][axis]['Cardinality'] for axis in range(dimensionality)]

Or maybe build_headers_from_cellset could also return dimensionality and cardinality?